src/main/java/core/option/UserColorsPanel.java

```diff
@@ -0,0 +1,340 @@
+		tableModel.setValueAt(color, row, 2);
+		tableModel.setValueAt(color.getDefaultValue(), row, 3);
+	}
+
+	/**
+	 * Reset color setting to given default setting
+	 * @param color Default color setting
+	 */
+	public void resetRow(HOColor color) {
+		colors.set(colorTable.getEditingRow(), color);
+		updateRow(color);
+	}
+
+	/**
+	 * Create label displaying the color name (1st column of the settings table)
+	 * The provided technical color name is translated to the select language.
+	 * This value is used as tool tip too.
+	 * @param colorName Color name
+	 * @return JLabel
+	 */
+	private JLabel createNameLabel(HOColorName colorName) {
+		String text = colorName!=null?colorName.toString():"";
+		var label = new JLabel(text);
+		label.setToolTipText(text);
+		return label;
+	}
+
+	/**
+	 * Get selected theme name
+	 * @return Theme name
+	 */
+	public String getSelectedTheme(){
+		return UserParameter.temp().skin;
+	}
+
+	/**
+	 * action
+	 */
+	public void actionPerformed(ActionEvent arg0) {
+		if (arg0.getSource() == skins) {
+			UserParameter.temp().skin = (String) skins.getSelectedItem();
+			initData(UserParameter.instance().skin);
+			// TODO change the look and feel dynamically
+			OptionManager.instance().setRestartNeeded();
+		}
+
+	}
+
+	/**
+	 * Store the edited color settings
+	 */
+	public void storeChangedColorSettings() {
+		for (var color : colors) {
+			var theme = getSelectedTheme();
+			var origValue = HOColor.getHOColor(color.getHOColorName(), theme);
+			assert origValue != null;
+			if (AreDifferentColors(origValue, color) ||
+					AreDifferentColors(origValue.getDefaultValue(), color.getDefaultValue())) {
+				if (!color.getTheme().equals("default")) {
+					color.setTheme(theme);
+					DBManager.instance().storeHOColor(color);
+				}
+				else {
+					// reset default
+					DBManager.instance().deleteHOColor(origValue);
+				}
+				HOColor.addColor(color);
+				OptionManager.instance().setRestartNeeded();
+			}
+		}
+	}
+
+	/**
+	 * Compare two color settings
+	 * @param c1 Color setting, may be null
+	 * @param c2 Color setting, may  be null
+	 * @return True, if the settings are different
+	 */
+	private boolean AreDifferentColors(HOColor c1, HOColor c2) {
+		if (c1 == null || c2 == null) return c1 != c2;
+		return AreDifferent(c1.getColor(), c2.getColor()) || AreDifferent(c1.colorReference(), c2.colorReference());
+	}
+
+	/**
+	 * Compare two objects
+	 * @param o1 Object, may  be null
+	 * @param o2 Object, may  be null
+	 * @return True, if the objects are different
+	 */
+	private boolean AreDifferent(Object o1, Object o2) {
+		if (o1 == o2) return false;
+		if (o1 == null) return true;
+		return !o1.equals(o2);
+	}
+
+	/**
+	 * Get the color value of the color setting
+	 * If no value is set, the corresponding reference is searched in the internal setting list.
+	 * @param currentColor Color setting
+	 * @return Color value
+	 */
+	public Color getColor(HOColor currentColor) {
```

Comment:
This also probably belongs in `HOColor`.

Reply:
this method works with the copied colors of the panel. it has to stay in the UserColorsPanel.

Reply:
Ok, I must misunderstand what `colorReference` does then.

This method could still be split into 2, one that's in `HOColor`, which returns the `getColor` colour, and a second argument which is a reference colour by default if there is no `getColor`, and then the panel-specific code, which I believe is that `colors.stream()...` part.  But it's only a suggestion, to segregate responsibilities.  I'll leave it to your judgment. ;-)

Reply:
if I understand your suggestion correctly, streaming would have to be done in any case, even if the color in currentColor should have a value.
I don't think that's so good.

Reply:
Now I got it